sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -459,7 +459,7 @@ pi_result _pi_context::finalize() {
   // There could be some memory that may have not been deallocated.
   // For example, zeEventPool could be still alive.
   std::lock_guard<std::mutex> NumEventsLiveInEventPoolGuard(
-      NumEventsLiveInEventPoolMutex, std::adopt_lock);
+      NumEventsLiveInEventPoolMutex);
```

Comment:
This indeed looks like copy-paste error to me, and the fix looks good.
Still, I wonder how this worked before. Basically, all the tests, including non-ESIMD should have been failing the same way, as it does not seem any thread can come here holding the `NumEventsLiveInEventPoolMutex`. Leaving to @smaslov-intel to review/approve.

Reply:
I think it wasn't noticed earlier because it only appears on debug builds on Windows.